miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/database/MiniAppSecureDatabase.kt

```diff
@@ -435,7 +435,10 @@ internal class MiniAppSecureDatabase(
             throw e
         } finally {
             finalize()
+            if (!isDeleted && items.size > 1)
+                isDeleted = !isDeleted
         }
+
         return isDeleted
```

Comment:
@gopi-karmakar why not just like below 🤔 ?

```
if (items.size > 1)
    isDeleted = true
```

Reply:
Yeah That is an obvious one but we will enter in to this condition only if the `isDeleted` is `false` so assigning with `!isDeleted`. It's something what I have noticed in the professional open source coding. Avoid hard coding as much as possible and rather taking the variables value in the runtime to be precisely sure that if the logic is working fine then the variable value should be what is expected.

Reply:
@gopi-karmakar what is the reason of this check, `items.size > 0` might always be true right? as i can not see any remove item operation. can you please explain for which scenario is condition will happen?

so what i understood  `isDeleted` is true if it can delete an item, after the delete operation is done and `isDeleted` is still false may be for some exceptions, you just simply made it true and return it.it looks like even it fails to delete items it still returns true.

Reply:
i understood after analyzing the code, both @laubengaier and @gopi-karmakar suggestion will work, but in this way there is one assertion less if `isDeleted` is already true, thanks.